### PR TITLE
FILTER-4: Add more tests and refactor for Datafilter UI

### PR DESCRIPTION
**Pull request by PremnishaTW**
FILTER-4: Add more tests and refactor for Datafilter UI

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/datafilter/web/UserFormSubmissionHandler.java [hunk 1]
```diff
@@ -41,35 +41,35 @@
...
+			throws IOException, ServletException {
+
 		//Process form data from legacy UI
 		chain.doFilter(request, response);
-		
+
 		if (((HttpServletRequest) request).getMethod().equals("POST")
-		        && ((HttpServletResponse) response).getStatus() == 200) {
-			
+				&& ((HttpServletResponse) response).getStatus() < 400) {
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/datafilter/web/UserFormSubmissionHandler.java:55]: Why change to <400?

**Comment by wluyima** [comment]
Please make sure to use the openmrs formatter

File: omod/src/main/java/org/openmrs/module/datafilter/web/UserFormSubmissionHandler.java [hunk 2]
```diff
@@ -41,35 +41,35 @@
 
 @Component("userFormSubmissionHandler")
 public class UserFormSubmissionHandler {
-
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/datafilter/web/UserFormSubmissionHandler.java:44]: Why do you have these format changes? Please install the openmrs code formatter in your IDE and use that to format your code, the formatter is in the tools sub module of openmrs core

File: omod/src/main/java/org/openmrs/module/datafilter/web/UserFormSubmissionHandler.java (hunk 1 above)
Comment (PremnishaTW) [reply omod/src/main/java/org/openmrs/module/datafilter/web/UserFormSubmissionHandler.java:55]: In legacy UI we have only handled setting status on error and by default we dont get 200 we get 302 redirect if user is created successfully

File: omod/src/main/java/org/openmrs/module/datafilter/web/UserFormSubmissionHandler.java (hunk 2 above)
Comment (PremnishaTW) [reply omod/src/main/java/org/openmrs/module/datafilter/web/UserFormSubmissionHandler.java:44]: I have installed the Openmrs formatter that has been mentioned last time from openmrs core in IDE. I did format using that and see the above changes. 
Please let me know if I am missing something

image

**Pull request merged**